Makefile

```diff
@@ -73,6 +73,28 @@ downstream_local:
 	    rm -fr ${DOWNSTREAM_NAME}-${VERSION}.tar.gz
 	git checkout ${UPSTREAM_NAME}.spec
 
+update_template:
+	# Clone from upstream and put required location.
+	curl -sL https://github.com/openshift/nodejs-ex/raw/master/openshift/templates/nodejs-mongodb.json > services/openshift/templates/common/nodejs-mongodb.json
+	curl -sL https://github.com/openshift/nodejs-ex/raw/master/openshift/templates/nodejs.json > services/openshift/templates/common/nodejs.json
+	curl -sL https://github.com/openshift/cakephp-ex/raw/master/openshift/templates/cakephp.json > services/openshift/templates/common/cakephp.json
+	curl -sL https://github.com/openshift/cakephp-ex/raw/master/openshift/templates/cakephp-mysql.json > services/openshift/templates/common/cakephp-mysql.json
+	curl -sL https://github.com/openshift/origin/raw/master/examples/jenkins/jenkins-ephemeral-template.json > services/openshift/templates/common/jenkins-ephemeral-template.json
+	curl -sL https://github.com/openshift/origin/raw/master/examples/jenkins/jenkins-persistent-template.json > services/openshift/templates/common/jenkins-persistent-template.json
+	curl -sL https://github.com/openshift/origin/raw/master/examples/jenkins/master-slave/jenkins-slave-template.json > services/openshift/templates/common/jenkins-slave-template.json
+	curl -sL https://github.com/jboss-openshift/application-templates/raw/master/eap/eap64-mysql-persistent-s2i.json > services/openshift/templates/common/eap64-mysql-persistent-s2i.json
+	curl -sL https://github.com/jboss-openshift/application-templates/raw/master/webserver/jws30-tomcat7-mysql-persistent-s2i.json > services/openshift/templates/common/jws30-tomcat7-mysql-persistent-s2i.json
+	curl -sL https://github.com/jboss-openshift/application-templates/raw/master/eap/eap64-basic-s2i.json > services/openshift/templates/common/eap64-basic-s2i.json
+	curl -sL https://github.com/jboss-openshift/application-templates/raw/master/jboss-image-streams.json > services/openshift/templates/common/jboss-image-streams.json
+	curl -sL https://github.com/openshift/origin/raw/master/examples/image-streams/image-streams-centos7.json > services/openshift/templates/adb/image-streams-centos7.json
+	curl -sL https://github.com/openshift/origin/raw/master/examples/image-streams/image-streams-rhel7.json > services/openshift/templates/cdk/image-streams-rhel7.json
+
+update_autocomplete:
+	# Clone latest autocomplete script and put required location
+	curl -sL https://github.com/openshift/origin/raw/master/contrib/completions/bash/oadm > bash_completions/oadm
+	curl -sL https://github.com/openshift/origin/raw/master/contrib/completions/bash/oc > bash_completions/oc
+	curl -sL https://github.com/openshift/origin/raw/master/contrib/completions/bash/openshift > bash_completions/openshift
+
```

Comment:
@praveenkumar  I believe we can remove those auto completion files either now or in the future because with new versions of **oc** you can do in your .{zsh,bsh}rc:
```zsh
source <(oc completion zsh)
```